app/pipedrive/_process.py

```diff
@@ -15,11 +15,19 @@ async def _process_pd_organisation(
     TODO: If we can't match the company by it's hermes_id, we should really try and match it by name also. However, as
     of now it should be impossible to create a company in Pipedrive that already exists in Hermes as the only other
     two ways to create a company (from TC2 and the Callbooker) always create the new company in PD.
+
+    if a company is not found in a_validate, then we will also check if it already exists in the db by matching the
+    pd_org_id
     """
     # Company has been set here by Org.a_validate, as we have a custom field `hermes_id` linking it to the Company
     current_company = getattr(current_pd_org, 'company', None) if current_pd_org else None
     old_company = getattr(old_pd_org, 'company', None) if old_pd_org else None
-    company = current_company or old_company
+
+    try:
+        existing_company = await Company.get(pd_org_id=current_pd_org.id) if current_pd_org else None
+    except Exception:
```

Comment:
We NEVER catch `Exception`, it's too vague. What error are you trying to stop here?

Reply:
changed to `DoesNotExist`